tests/testthat/test-tab_afun_cfun.R

```diff
@@ -0,0 +1,233 @@
+                           SEX == .spl_context$value[3])
+                
+                # All strata is add_overall_level -> filter not needed
+                if (.spl_context$value[[2]] != "All Strata") {
+                alt_df1b <- alt_df1b %>% 
+                    filter(STRATA1 == .spl_context$value[[2]])
+                alt_df2b <- alt_df2b %>% 
+                    filter(STRATA1 == .spl_context$value[[2]])
+                }
+                
+                # This would break the tests if no match
+                stopifnot(nrow(alt_df1) == nrow(alt_df1b))
+                stopifnot(nrow(alt_df1) == nrow(alt_df1c))
+                stopifnot(nrow(alt_df2) == nrow(alt_df2b))
+                stopifnot(nrow(alt_df2) == nrow(alt_df2c))
+                
+                # General info
+                stopifnot(.all_col_counts[[.spl_context$cur_col_id[[1]]]] == .N_col)
+                stopifnot(identical(.all_col_exprs[[.spl_context$cur_col_id[[1]]]],
+                              .spl_context$cur_col_expr[[3]]))
+                
+                # Fin needed output 
+                in_rows("n" = c(nrow(alt_df1c), 
+                                nrow(alt_df2c)), 
+                        .formats = "xx - xx")
+            }
+        }
+    }
+    
+    lyt <- basic_table(show_colcounts = TRUE) %>% 
+        split_cols_by("ARM", split_fun = add_combo_levels(combodf, first = TRUE)) %>%
+        split_cols_by("COUNTRY", split_fun = keep_split_levels(c("CHN", "USA"))) %>%
+        split_rows_by("STRATA1", split_fun = add_overall_level("All Strata")) %>%
+        split_rows_by("SEX", split_fun = keep_split_levels(c("F", "M"))) %>%
+        add_overall_col("All Patients 2") %>% 
+        analyze(vars = "BMRKR1", afun = n_wrapper_alt_df(ex_adsl))
+    
+    # NB: If you add keep_levels = c("all_X") to add_combo_levels the other 
+    #     column expressions are missing -> Expected!
+    
+    tbl <- lyt %>% build_table(DM, alt_counts_df = ex_adsl)
+    
+    expect_silent(cbind_rtables(tbl, tbl))
+    expect_silent(rbind(tbl, tbl))
+    
+    spl_ctx_cnt <- lapply(seq(8, nrow(tbl), 5), function(x) tbl[x, 2, drop = TRUE])
+    
+    nrow_manual <- lapply(sort(unique(ex_adsl$STRATA1)), function(x) {
+        tmp_strata <- ex_adsl %>% filter(STRATA1 == x, SEX == "F", COUNTRY == "USA")
+        sapply(list(tmp_strata %>% filter(ARM == "A: Drug X"),
+                 tmp_strata %>% filter(ARM == "C: Combination")), nrow)
+    })
+    
+    expect_identical(nrow_manual, spl_ctx_cnt)
+})
+
+test_that("Error localization for missing split variable when done in alt_count_df", {
+    # Error we want to happen
+    afun_tmp <- function(x, .alt_df, ...) mean(x)
+    lyt_col <- basic_table() %>% split_cols_by("ARMCD") %>% analyze("BMRKR1", afun = afun_tmp)
+    lyt_row <- basic_table() %>% split_rows_by("ARMCD") %>% analyze("BMRKR1", afun = afun_tmp)
+    expect_error(lyt_col %>% build_table(ex_adsl, alt_counts_df = DM))
```

Comment:
Error here is as below. I don't think it should mention `df` since the problem is `alt_counts_df`? 
Also is there a way to make it read nicer like "alt_counts_df:  variable\\(s\\) \\[ARMCD\\] not present in data"?
```
Error in FUN(X[[i]], ...) :
alt_counts_df (or df) appears incompatible with column-split structure. Offending column subset expression: (!is.na(ARMCD) & ARMCD %in% "ARM A")
Original error message: object 'ARMCD' not found
```

Reply:
I think we should mention columns rather than simplifying all the way down to what @anajens mentioned, because it is indicating where the problem occurred. 

Other than that, honestly these checks should be happening at build_table time rather than in the guts of tabulation, I think. For now its probably fine though.